app/scripts/metamask-controller.js

```diff
@@ -2028,6 +2068,16 @@ export default class MetamaskController extends EventEmitter {
         });
       }
 
+      case KEYRING_TYPES.QR: {
+        return new Promise((_, reject) => {
```

Comment:
Instead of the `new Promise()` usage, we can simply do:

```
return Promise.reject(new Error('QR hardware does not support eth_getEncryptionPublicKey.'));
```